modules/indexer/code/bleve/bleve.go

```diff
@@ -260,17 +260,28 @@ func (b *Indexer) Search(ctx context.Context, opts *internal.SearchOptions) (int
 	var (
 		indexerQuery query.Query
 		keywordQuery query.Query
+		contentQuery query.Query
 	)
 
 	pathQuery := bleve.NewPrefixQuery(strings.ToLower(opts.Keyword))
 	pathQuery.FieldVal = "Filename"
 	pathQuery.SetBoost(10)
 
-	contentQuery := bleve.NewMatchQuery(opts.Keyword)
-	contentQuery.FieldVal = "Content"
-
-	if opts.IsKeywordFuzzy {
-		contentQuery.Fuzziness = inner_bleve.GuessFuzzinessByKeyword(opts.Keyword)
+	keywordAsPhrase, isPhrase := internal.ParseKeywordAsPhrase(opts.Keyword)
+	if isPhrase {
+		q := bleve.NewMatchPhraseQuery(keywordAsPhrase)
+		q.FieldVal = "Content"
+		if opts.IsKeywordFuzzy {
+			q.Fuzziness = inner_bleve.GuessFuzzinessByKeyword(keywordAsPhrase)
+		}
+		contentQuery = q
+	} else {
+		q := bleve.NewMatchQuery(opts.Keyword)
```

Comment:
It seems to me like an unintended regression from https://github.com/go-gitea/gitea/pull/32210/files#diff-e1866799c5d7d231d9df896b84d4a9fa54d404d150e26ac6b205eaf0fea3790bL243-R270 that the default changed from NewMatchQuery to NewMatchPhraseQuery. When searching for something like `swaggerParameterBodies` it is currently searching separately on `swagger`, `parameter` and `bodies`.

So I wonder if it shouldn't simply be switched back to NewMatchPhraseQuery, without requiring the quotes?

Reply:
I agree (I seldom use the "code search" feature, and just tried to help to make this PR's code more general).

@jpraet @lunny  Would you like to propose some PRs to fix the regression? And maybe make the elasticsearch have the same behavior (I will improve the git grep search later). Thank you.

Reply:
@bsofiato can you confirm the switch from MatchPhraseQuery to MatchQuery for bleve in #32210 was unintentional?